pkg/ebpf/state.go

```diff
@@ -156,10 +178,41 @@ func (ns *networkState) Connections(id string, latestTime uint64, latestConns []
 	// Flush closed connection map and stats
 	ns.clients[id].closedConnections = map[string]ConnectionStats{}
 	ns.determineConnectionIntraHost(conns)
+	ns.storeDNSStats(dnsStats)
+	ns.addDNSStats(id, conns)
 
 	return conns
 }
 
+func (ns *networkState) addDNSStats(id string, conns []ConnectionStats) {
+	seen := make(map[dnsKey]struct{}, len(conns))
+	for i := range conns {
+		conn := &conns[i]
+		if conn.DPort != 53 {
+			continue
+		}
+		key := dnsKey{
+			serverIP:   conn.Dest,
+			clientIP:   conn.Source,
+			clientPort: conn.SPort,
+			protocol:   conn.Type,
+		}
+
+		if _, alreadySeen := seen[key]; alreadySeen {
+			ns.telemetry.dnsPidCollisions++
```

Comment:
why are we calling this pid collisions?  it looks like it is colliding on the (srcip, dstip, sport, dport) combo?

Reply:
I tried to be consistent with how we deal with a similar situation when processing TCP stats which also does not have PID information: https://github.com/DataDog/datadog-agent/blob/master/pkg/ebpf/tracer.go#L501

I think the idea is that different PIDs are ending up being mapped to the same value - so they are _colliding_ 😕 .